src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -463,6 +519,44 @@ public static long ToInt64(ReadOnlySpan<byte> value)
             return Unsafe.ReadUnaligned<long>(ref MemoryMarshal.GetReference(value));
         }
 
+        /// <summary>
+        /// Returns a 128-bit signed integer converted from sixteen bytes at a specified position in a byte array.
+        /// </summary>
+        /// <param name="value">An array of bytes.</param>
+        /// <param name="startIndex">The starting position within <paramref name="value"/>.</param>
+        /// <returns>A 128-bit signed integer formed by sixteen bytes beginning at <paramref name="startIndex"/>.</returns>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="startIndex"/> is greater than or equal to the length of <paramref name="value"/> minus 15,
+        /// and is less than or equal to the length of <paramref name="value"/> minus 1.
+        /// </exception>
+        /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startIndex"/> is less than zero or greater than the length of <paramref name="value"/> minus 1.</exception>
+        public static Int128 ToInt128(byte[] value, int startIndex)
+        {
+            if (value == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
```

Comment:
ArgumentNullException.ThrowIfNull, etc?

Reply:
Just following the existing convention in the rest of the file. Updating everything to use `ThrowIfNull` could/should be done in a separate PR so things stay consistent.